website/server/models/group.js

```diff
@@ -344,6 +433,8 @@ schema.statics.toJSONCleanChat = async function groupToJSONCleanChat (group, use
     await getGroupChat(group);
   }
 
+  group = _translateSystemMessages(group, user);
```

Comment:
this should be moved after the toJSON conversion or it'll modify the mongoose object directly which we don't want to do

Reply:
Updating the `group` after the `let toJSON` isn't meaningless?
The `group` variable isn't used anymore after `let toJSON = group.toJSON();`.

Reply:
If you modify the group object before serializing it to json it means that you're changing the mongoose class representing the group (so the in memory representation of the db data) which we don't want to do since translated messages should never end up in the db.

Instead we want to first convert it to json `let toJSON = group.toJSON(); ` and then translate that object with `_translateSystemMessages(toJSON, user); `

Reply:
But after serializing it into JSON, it is not saved to the database. For testing, I changed the language here and the database didn't change at all.
All is done just to show the messages to the user, there is no save involved.

And if there is a case where it would be saved, why did you ask me to make the `translateSystemMessages` to a class method?
The `group` object wouldn't be needed anymore...

By the way, I already changed the method to a class method.

Reply:
And If there is such possibility of `group` being saved, I think the call to `getGroupChat` is a bigger problem than my call to translate the messages.